vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/executor/listeners/video/ScreenRecorderTestRunListener.kt

```diff
@@ -62,43 +67,53 @@ class ScreenRecorderTestRunListener(
     }
 
     override suspend fun testAssumptionFailure(test: TestIdentifier, trace: String) {
-        pullVideo(test.toTest())
+        //Treating as failure
+        hasFailed = true
     }
 
     override suspend fun testEnded(test: TestIdentifier, testMetrics: Map<String, String>) {
         pullVideo(test.toTest())
+        lastRemoteFile = null
     }
 
     override suspend fun testRunFailed(errorMessage: String) {
-        val join = measureTimeMillis {
-            logger.trace { "cancel" }
-            supervisorJob?.cancelAndJoin()
+        try {
+            if (device.verifyHealthy()) {
```

Comment:
device might not be available. we have to quickly establish if it's working, otherwise the batch result will be delayed